pandas/indexes/base.py

```diff
@@ -1889,7 +1889,20 @@ def get_loc(self, key, method=None, tolerance=None):
                 raise ValueError('tolerance argument only valid if using pad, '
                                  'backfill or nearest lookups')
             key = _values_from_object(key)
-            return self._engine.get_loc(key)
+            try:
+                return self._engine.get_loc(key)
+            except KeyError:
+
+                # coerce floats to ints if they are equal
+                # and we are not a floating index
+                if is_float(key) and not self.is_floating():
+                    try:
+                        ckey = int(key)
+                        if ckey == key:
```

Comment:
This should actully be done inside the indexing engine......will make an issure for it